.github/workflows/post-merge-deploy.yml

```
@@ -0,0 +1,48 @@
+name: Package for Deployment
+on:
+  workflow_dispatch:
+  push:
+    branches: [main]
+
+jobs:
+  deploy:
+    name: Package for build
+    runs-on: ubuntu-latest
+    timeout-minutes: 15
+    environment: ${{ matrix.environment }}
+    strategy:
+      matrix:
+        environment: [development, build]
+    concurrency:
+      group: post-merge-${{ matrix.environment }}
+      cancel-in-progress: false
+    env:
+      AWS_REGION: eu-west-2
+    permissions:
+      id-token: write
+      contents: read
+    steps:
+      - name: Check out repository code
+        uses: actions/checkout@v4
+
+      - name: Assume temporary AWS role
+        uses: aws-actions/configure-aws-credentials@v4
+        with:
+          role-to-assume: ${{ vars.GH_ACTIONS_ROLE_ARN }}
+          aws-region: ${{ env.AWS_REGION }}
+
+      - name: Build SAM app
+        uses: govuk-one-login/github-actions/sam/build-application@cd7d35dde348251237efbbaee5345e95adef0321
+        id: build
+        with:
+          template: infrastructure/template.yaml
+          cache-name: otg-smoke-tests
+          pull-repository: false
+
+      - name: Deploy SAM app
```

```suggestion
      - name: Upload to S3
```
This step only uploads to S3 to be picked up for deployment for the pipeline